Update Gradle wrapper to 4.10.2

friederbluemle opened the pull request:
This updates Gradle wrapper to `4.10.2` (latest version as of Oct 3, 2018), and also adds a `.gitignore` file to not show build output as untracked files.

sherter commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,5 @@
-#Fri Apr 14 20:53:26 CEST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.6-all.zip
```
I'd like to continue using the `bin` variant (it's half the size)

friederbluemle replied:
Sure, I changed it to the `bin` variant.

friederbluemle commented:
@sherter Sorry for the slow reply - I updated and rebased this PR. Gradle `4.10.2` is out now and I amended the commit and the PR description.

Regarding changes in `gradle-wrapper.jar` - On a high level (just looking at the difference of class files between 4.2.1 and 4.10.2), it looks like the main changes were in the command line parser, as well as the download proxy authenticator.

I definitely understand and agree that unnecessary repo size increase should be avoided. The file size is 55k, and there is about 1 year of development between 4.2.1 and 4.10.2. Regarding the distribution URL, I changed it back to `-bin`.

sherter commented:
Updated to 4.10.2 in d3a3a4929ef9cf0fb4070b159bf9581e5fae2b10
